pkg/healthcheck/healthcheck.go

```diff
@@ -255,3 +270,37 @@ func (hc *HealthChecker) RunChecks(observer checkObserver) bool {
 func (hc *HealthChecker) PublicAPIClient() pb.ApiClient {
 	return hc.apiClient
 }
+
+func validatePods(pods []v1.Pod) error {
+	statuses := make(map[string][]v1.ContainerStatus)
```

Comment:
it feels like we should be able to use `ContainersReady` from the [pod conditions](https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-readiness-gate) for this but I cant actually figure out how... the documentation is not really clear

Reply:
Hmm, yeah, looks like pod conditions are new in 1.11 -- we could consider switching to those eventually. In versions prior to 1.11, `ContainerStatus.Ready` will tell us if the pod is ready for traffic.